### PR TITLE
Ethernet/net mgmt fixes

### PR DESCRIPTION
**Pull request by tbursztyka**
Layer code mask is 0x7FF so obviously 0x802 is not valid (as it will
always set the synchronous bit).

### CODE REVIEW

**Review by jukkar — APPROVED** [review]

**Review by lpereira — APPROVED** [review]

**Comment by lpereira** [comment]
It would be nice if these management codes were generated automatically somehow (e.g. like how Linux creates ioctl() codes); I wonder if we have more of similar issues laying around.

**Comment by tbursztyka** [comment]
@nashif Can you apply that one? Thanks.

**Comment by tbursztyka** [comment]
@nashif wait, DNM yet.

File: subsys/net/ip/net_mgmt.c [hunk 1]
```diff
@@ -56,7 +56,7 @@ static inline void mgmt_push_event(u32_t mgmt_event, struct net_if *iface,
 	ARG_UNUSED(length);
 #endif /* CONFIG_NET_MGMT_EVENT_INFO */
 
-	k_sem_take(&net_mgmt_lock, K_FOREVER);
+	k_sem_take(&net_mgmt_lock, k_is_in_isr() ? K_NO_WAIT : K_FOREVER);
```
Comment (pfalcon) [inline subsys/net/ip/net_mgmt.c]: From https://github.com/zephyrproject-rtos/zephyr/pull/8027#pullrequestreview-125533551 :

What it k_sem_take() fails with K_NO_WAIT?

Comment (tbursztyka) [reply subsys/net/ip/net_mgmt.c]: Actually, I'll get rid of that patch: net mgmt is a user notifying mechanism, not a core one so it should not be used in ISR mode.

**Pull request merged**